transactions/pendingtxtracker.go

```diff
@@ -59,8 +62,10 @@ const (
 )
 
 type TxIdentity struct {
-	ChainID common.ChainID `json:"chainId"`
-	Hash    eth.Hash       `json:"hash"`
+	ChainID          common.ChainID                     `json:"chainId"`
+	Hash             eth.Hash                           `json:"hash"`
+	SendDetails      *responses.SendDetails             `json:"sendDetails"`
```

Comment:
Not good. TxIdentity is supposed to be a very generic super light struct used to uniquely identify a transaction (so, chainID and hash). And PendingTxTracker is supposed to be a generic and single-responsibility (though it's more complex than it should be) Tx status tracker. I'd rather keep it that way and not add a bunch of data to the type/dependencies to the tracker.
IMO, if the client needs more data, better add some API endpoint like "GetRouteDetailsForTxIdentity" and have the client fetch it. Otherwise, create some other event type with this chunky payload and emit it separately.

Reply:
There is no use case where `TxIdentity` was sent as a payload of a signal, but I agree, based on the type name it's too much for that type, that's why I added a new type. Also, wherever we want to update the client about the tx change, we need to provide the full context (not only txHash and chainId) so that the client knows what change occurred actually and what to do.

To avoid confusion, I moved `SendDetails` and `SentTransactions` to a new `TxDetails` type that both `PendingTxUpdatePayload` and `StatusChangedPayload` have now.